add docs for setting async_insert_max_query_number (#46666)
ClickHouse/ClickHouse: docs/en/operations/settings/settings.md

Default value: `100000`.

### async_insert_max_query_number {#async-insert-max-query-number}

> The maximum number of insert query in per block before being inserted. It takes effect only if [async_insert_deduplicate](#settings-async-insert-deduplicate) is enabled.

Reviewer: ```suggestion
The maximum number of insert queries per block before being inserted. This setting takes effect only if [async_insert_deduplicate](#settings-async-insert-deduplicate) is enabled.
```
